lib/archethic/contracts/interpreter/library/common/http_impl.ex

```diff
@@ -58,74 +61,85 @@ defmodule Archethic.Contracts.Interpreter.Library.Common.HttpImpl do
 
   @tag [:io]
   @impl Http
-  def request_many(uris) do
-    if check_too_many_calls() do
-      raise Library.Error, message: "Http module got called more than once"
-    end
+  def request_many(requests) do
+    check_too_many_calls()
 
-    uris_count = length(uris)
+    requests_count = length(requests)
 
-    if uris_count > 5 do
+    if requests_count > 5 do
```

Comment:
Suggestion: to simplify the codebase, you could use a guard upfront . And let pattern matching do the rest for the happy path.